diff --git a/shell/linux/Makefile b/shell/linux/Makefile
index 289836067..86365d5bd 100644
--- a/shell/linux/Makefile
+++ b/shell/linux/Makefile
@@ -42,6 +42,12 @@ ifeq (,$(platform))
     else ifeq ($(ARCH), $(filter $(ARCH), x86_64 AMD64 amd64))
         platform = x64
     else ifneq (,$(findstring aarch64,$(ARCH)))
+ RK_BOARD = $(shell cat /sys/firmware/devicetree/base/model | cut -f2 -d" ")
+ifneq (,$(findstring RockPro64,$(RK_BOARD)))
+    platform = rockpro64
+else ifneq (,$(findstring Rock64,$(RK_BOARD)))
+    platform = rock64
+endif
         HARDWARE = $(shell grep Hardware /proc/cpuinfo)
         ifneq (,$(findstring Vero4K,$(HARDWARE)))
             platform = vero4k
@@ -233,6 +239,22 @@ else ifneq (,$(findstring vero4k,$(platform)))
     USE_GLES := 1
     USE_SDL := 1

+# RockPro64
+else ifneq (,$(findstring rockpro64,$(platform)))
+    MFLAGS += -O2 -marm -march=armv8-a+crc -mtune=cortex-a72.cortex-a53 -mfloat-abi=hard -funsafe-math-optimizations -ftree-vectorize -funroll-loops
+    ASFLAGS += -marm -march=armv8-a+crc -mtune=cortex-a72.cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard
+    CFLAGS += -D TARGET_BEAGLE -D TARGET_LINUX_ARMELv7 -DARM_HARDFP
+    USE_GLES := 1
+    USE_SDL := 1
+
+# Rock64
+else ifneq (,$(findstring rock64,$(platform)))
+    MFLAGS += -O2 -marm -march=armv8-a+crc -mtune=cortex-a53 -mfloat-abi=hard -funsafe-math-optimizations -ftree-vectorize -funroll-loops
+    ASFLAGS += -marm -march=armv8-a+crc -mtune=cortex-a72.cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard
+    CFLAGS += -D TARGET_BEAGLE -D TARGET_LINUX_ARMELv7 -DARM_HARDFP
+    USE_GLES := 1
+    USE_SDL := 1
+
 else
     $(error Unknown platform)
 endif
